From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct ehci_regs
Patch-mainline: never, kabi

In 3.12.62, commit 6fcbf62da1b054958dffbf039fe42c0743072ceb (USB:
EHCI: declare hostpc register as zero-length array), upstream commit
7e8b3dfef16375dbfeb1f36a83eb9f27117c51fd moved one u32 from hostpc to
the following reserved5 in struct ehci_regs. It fooled kABI checker
and it thinks the structure has changed.

The structure actually did not change, so pretend that the usual way,
by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/usb/ehci_def.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/include/linux/usb/ehci_def.h
+++ b/include/linux/usb/ehci_def.h
@@ -178,11 +178,16 @@ struct ehci_regs {
  * PORTSCx
  */
 	/* HOSTPC: offset 0x84 */
+#ifdef __GENKSYMS__
+	u32		hostpc[1];	/* HOSTPC extension */
+	u32		reserved5[16];
+#else
 	u32		hostpc[0];	/* HOSTPC extension */
 #define HOSTPC_PHCD	(1<<22)		/* Phy clock disable */
 #define HOSTPC_PSPD	(3<<25)		/* Port speed detection */
 
 	u32		reserved5[17];
+#endif
 
 	/* USBMODE_EX: offset 0xc8 */
 	u32		usbmode_ex;	/* USB Device mode extension */
